### PR TITLE
Add limit check to add_event

### PR DESCRIPTION
**Pull request by kohlerca**
When using the `add_event` function to add events to one or more plots (e.g., the output of `rasterplot_rates`), if the Axes limits were changed and the Event object includes events out of the boundaries, they will be plotted regardless, leading to an unpleasant visualization in notebooks (saving the figure to a file seems to work).

```
import quantities as pq
from elephant.spike_train_generation import StationaryPoissonProcess
from viziphant.rasterplot import rasterplot_rates
from viziphant.events import add_event
import matplotlib.pyplot as plt
import neo

spiketrains = StationaryPoissonProcess(rate=20 * pq.Hz, t_stop= 2 * pq.s).generate_n_spiketrains(20)

event = neo.Event([0.25, 1, 1.75], units=pq.s, labels=['event_1', 'event_2', 'event_3'])

fig, ax = plt.subplots(figsize=(20,15))
ax = rasterplot_rates(spiketrains, ax=ax)
ax[0].set_xlim([0, 0.75])
ax[1].set_xlim([0, 0.75])
add_event([ax[1], ax[0]], event)
```

old

This PR fixes this by adding boundary checks (with respect to the last Axes in the list passed as the function `axes` parameter).

new

### CODE REVIEW

File: viziphant/events.py [hunk 1]
```diff
@@ -63,13 +63,18 @@ def add_event(axes, event, key=None, rotation=40):
...
-        if key is None:
-            label = event.labels[event_idx]
-        else:
-            label = event.array_annotations[key][event_idx]
-        for axis in axes:
-            axis.axvline(time, color='black')
-        axes[0].text(time, axes[0].get_ylim()[1], label,
-                     horizontalalignment='left',
-                     verticalalignment='bottom', rotation=rotation)
+        if x_lim_min <= time <= x_lim_max:
```
Comment (Moritz-Alexander-Kern) [inline viziphant/events.py:71]: Can we be sure that `x_lim_min < x_lim_max` always holds in this use case?

In the matplotlib documentation on  `.get_xlim` the note says:

_The x-axis may be inverted, in which case the left value will be greater than the right value._

See:
https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.get_xlim.html

Comment (Moritz-Alexander-Kern) [inline viziphant/events.py:71]: Optional:
(There might be edge cases, where an event is really close to being inside the limits but is discarded.)
- add a section "Notes" in the doctsring to inform users, that events may be discarded if they are outside the limits ?
- add a warning if events are discarded for this reason ?

Those are just ideas, what do you think ?

**Review by Moritz-Alexander-Kern — APPROVED** [review]
Thanks for this enhancement !

File: viziphant/events.py (hunk 1 above)
Comment (kohlerca) [reply viziphant/events.py:71]: If I understood correctly, the idea is to have a certain tolerance and check if there was an event close by outside the boundary and warn? From my perspective, the idea of the function is to discard events smaller than the lower limit and larger than the highest, as they are shown as those floating labels. So I am not sure about the benefit of such warnings.

Comment (Moritz-Alexander-Kern) [reply viziphant/events.py:76]: I agree, warning is not appropriate for this case.

Comment (Moritz-Alexander-Kern) [reply viziphant/events.py:76]: - add a section "Notes" in the doctsring to inform users, that events may be discarded if they are outside the limits ?

**Pull request merged**